examples/regression/script.jl

```diff
@@ -0,0 +1,196 @@
+# A recreation of <https://gpflow.readthedocs.io/en/master/notebooks/advanced/gps_for_big_data.html>
+
+# # Stochastic Variational Regression
+#
+# In this example, we show how to construct and train the stochastic variational
+# Gaussian process (SVGP) model for efficient inference in large scale datasets.
+# For a basic introduction to the functionality of this library, please refer to
+# the [User Guide](@ref).
+#
+# ## Setup
+
+using SparseGPs
+using Distributions
+using LinearAlgebra
+using IterTools
+
+using Plots
+default(; legend=:outertopright, size=(700, 400))
+
+using Random
+Random.seed!(1234)
+#md nothing #hide
+
+# ## Generate some training data
+#
+# The data generating function `g`
+
+function g(x)
+    return sin(3π * x) + 0.3 * cos(9π * x) + 0.5 * sin(7π * x)
+end
+
+N = 10000 # Number of training points
+x = rand(Uniform(-1, 1), N)
+y = g.(x) + 0.3 * randn(N)
+
+scatter(x, y; xlabel="x", ylabel="y", markershape=:xcross, markeralpha=0.1, legend=false)
+
+# ## Set up a Flux model
+#
+# We shall use the excellent framework provided by [Flux.jl](https://fluxml.ai/)
+# to perform stochastic optimisation. The SVGP approximation has three sets of
+# parameters to optimise - the inducing input locations, the mean and covariance
+# of the variational distribution `q` and the parameters of the
+# kernel.
+#
+# First, we define a helper function to construct the kernel from its parameters
+# (often called kernel hyperparameters), and pick some initial values `k_init`.
+
+using Flux
+
+function make_kernel(k)
+    return softplus(k[1]) * (SqExponentialKernel() ∘ ScaleTransform(softplus(k[2])))
+end
```

Comment:
Or how about:
```suggestion
function make_kernel(k_params)
    variance = softplus(k_params[1])
    lengthscale = softplus(k_params[2])
    return variance * with_lengthscale(SqExponentialKernel(), lengthscale)
end
```
(my preference)
or, if you prefer:

```suggestion
function make_kernel(k_params)
    variance = softplus(k_params[1])
    inv_lengthscale = softplus(k_params[2])
    return variance * (SqExponentialKernel() ∘ ScaleTransform(inv_lengthscale))
end
```